### DISCUSSION TITLE
Access to segmentation data in python

### DISCUSSION THREAD

**Discussion by simonvanvliet** [jeanollion/bacmman#9] [Ideas]
We have started using Bacmman for all our mother machine data analysis and are very happy with its performance. 
The python interface you provide is also super handy for integrating Bacmann in our data analysis workflow.

One thing we still miss however, is to have access to the segmented data (i.e. label images) from within python, such that we can easily implement additional cell measurements from within Python.

Would it be possible to add a function to PyBacmman to access this data?
Or could you provide some info on how data is stored in the .db files and how it can be accessed using standard sql packages?

**Comment by jeanollion** [comment]
Hello Simon, 
I'm glad to hear this works for you. 
The database used by bacmman is not SQL and to my knowledge there is no python API to access it. 
However you can extract label images as hdf5 files and use the h5py package to access them from python. This is described here: https://github.com/jeanollion/bacmman/wiki/FineTune-DistNet#extract-a-dataset-using-bacmman 
You only need to choose  `Labels` (and eventually `RawImage`) features. 

FYI I work as an independent researcher and bio-image analysis consultant, if you need a development I can provide it. Here is my website: www.sabilab.fr
Best, 
Jean Ollion
